src/main/java/org/folio/circulation/domain/CheckInLogRecord.java

```diff
@@ -0,0 +1,96 @@
+
+import java.util.UUID;
+
+import org.apache.commons.lang3.ObjectUtils;
+import org.joda.time.DateTime;
+
+import io.vertx.core.json.JsonObject;
+
+public class CheckInLogRecord {
+  private static final String ID = "id";
+  private static final String OCCURRED_DATE_TIME = "occurredDateTime";
+  private static final String ITEM_ID = "itemId";
+  private static final String CHECK_IN_SERVICE_POINT_ID = "checkInServicePointId";
+  private static final String PERFORMED_BY_USER_ID = "performedByUserId";
+
+  private final String id;
+  private final DateTime occurredDateTime;
+  private final String itemId;
+  private final String checkInServicePointId;
+  private final String performedByUserId;
+
+  private CheckInLogRecord(Builder builder) {
+    this.id = builder.id;
+    this.occurredDateTime = builder.occurredDateTime;
+    this.itemId = builder.itemId;
+    this.checkInServicePointId = builder.checkInServicePointId;
+    this.performedByUserId = builder.performedByUserId;
+  }
+
+  public static class Builder {
+    private String id = UUID.randomUUID().toString();
+    private DateTime occurredDateTime;
+    private String itemId;
+    private String checkInServicePointId;
+    private String performedByUserId;
+
+    public Builder withId(String id) {
+      this.id = id;
+      return this;
+    }
+
+    public Builder withOccurredDateTime(DateTime occurredDateTime) {
+      this.occurredDateTime = occurredDateTime;
+      return this;
+    }
+
+    public Builder withItemId(String itemId) {
+      this.itemId = itemId;
+      return this;
+    }
+
+    public Builder withCheckInServicePointId(String checkInServicePointId) {
+      this.checkInServicePointId = checkInServicePointId;
+      return this;
+    }
+
+    public Builder withPerformedByUserId(String performedByUserId) {
+      this.performedByUserId = performedByUserId;
+      return this;
+    }
+
+    public CheckInLogRecord build() {
+      if (!ObjectUtils.allNotNull(id, occurredDateTime, itemId,
+        checkInServicePointId, performedByUserId)) {
+
+        throw new IllegalStateException("All properties are required");
+      }
+
+      return new CheckInLogRecord(this);
+    }
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public JsonObject toJson() {
+    return new JsonObject()
+      .put(ID, id)
+      .put(OCCURRED_DATE_TIME, occurredDateTime.toString())
+      .put(ITEM_ID, itemId)
+      .put(CHECK_IN_SERVICE_POINT_ID, checkInServicePointId)
+      .put(PERFORMED_BY_USER_ID, performedByUserId);
+  }
+
+  public static CheckInLogRecord from(JsonObject json) {
+    return builder()
+      .withId(json.getString(ID))
+      .withOccurredDateTime(DateTime.parse(json.getString(OCCURRED_DATE_TIME)))
+      .withItemId(json.getString(ITEM_ID))
+      .withCheckInServicePointId(json.getString(CHECK_IN_SERVICE_POINT_ID))
+      .withPerformedByUserId(json.getString(PERFORMED_BY_USER_ID))
+      .build();
+  }
```

Comment:
When is there a need to read a check in log record from JSON?

Reply:
It is used to parse created record.

Reply:
When is the created record parsed? If it is only in tests, then I think this code should be test code rather than production code.

Reply:
It is production code - when receive POST reply - https://github.com/folio-org/mod-circulation/pull/481/files#diff-2889bbf3610c1e9f4d605de3dccdced8R26

Reply:
Why does the response need to be interpreted and returned to the client of the repository?

Reply:
In a fact, it is never used, but the ResponseInterpreter requires a mapper. Will investigate how to simplify this.

Reply:
The simplest mapper function that can be used is `Function.identity()` which maps to itself.